### ISSUE TITLE
ANR: VS Code lose response when search some file with specific regular expression

### ISSUE THREAD

**Issue by wysaid** [microsoft/vscode#145855]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: __(VS Code Insiders has the same problem)__
<img width="373" alt="image" src="https://user-images.githubusercontent.com/1430725/159657605-4e19f023-7573-4042-b1dd-ecc4e293ec3e.png"> <img width="268" alt="image" src="https://user-images.githubusercontent.com/1430725/159657658-0b53e855-72a4-4b1a-b108-65c58631c42c.png">

- OS Version:

Steps to Reproduce:

1. Open any cpp file like below:
    ```cpp
    #include "test.h"
    
    using namespace test;
    
    class TestClass
    {
        void someFunc()
        {
            TestType* alongNameVariableAAAAAAAAA;
            TestType* alongNameVariableBBBBBBB;
            TestType* alongNameVariableCCCCCC;
            TestType* alongNameVariableDDDDDD;
        }
    };
    ``` 
2. Open search window(ctrl + F for Windows or ⌘ + F for Mac)
3. Enable regular expression by click `the regular expression button`
    <img width="471" alt="Screen Shot 2022-03-23 at 16 41 22" src="https://user-images.githubusercontent.com/1430725/159658352-d0c72de8-290a-4737-9f6e-4a4a853173e1.png">
4. Paste the expression: ``#(if|inc|import)((.|\n| )*)#endif(.*)``
5. VS Code lost response after step 4.  Press `esc`, then press `ctrl + F` or `⌘+F` again if it doesn't happen.

- ## Bug Screen Shot

<img width="608" alt="image" src="https://user-images.githubusercontent.com/1430725/159659557-7a4b615c-c4e7-4547-ab51-2d8a73c5940b.png">

<img width="639" alt="image" src="https://user-images.githubusercontent.com/1430725/159659112-bc328083-a0d9-4c66-877e-253898b60358.png">

**Comment by IllusionMH** [comment]
/duplicate of #30874
Or at least related as `(.|\n| )*` causes problems where `.` can match both `\n` and ` `.

`[\s\S\n]*` is better replacement
